pkg/prometheus/promcfg.go

```diff
@@ -1871,12 +1878,52 @@ func (cg *ConfigGenerator) generateRemoteWriteConfig(
 		}
 
 		if spec.AzureAD != nil {
-			azureAd := yaml.MapSlice{
-				{
-					Key: "managed_identity", Value: yaml.MapSlice{
+			if spec.AzureAD.ManagedIdentity.ClientID == "" && spec.AzureAD.OAuth == nil {
+				return yaml.MapItem{}, fmt.Errorf("must provide Azure Managed Identity or Azure OAuth in the Azure AD config")
+			}
+			if spec.AzureAD.ManagedIdentity.ClientID != "" && spec.AzureAD.OAuth != nil {
+				return yaml.MapItem{}, fmt.Errorf("cannot provide both Azure Managed Identity and Azure OAuth in the Azure AD config")
+			}
+			azureAd := yaml.MapSlice{}
+
+			if spec.AzureAD.ManagedIdentity.ClientID != "" {
+				azureAd = append(azureAd,
+					yaml.MapItem{Key: "managed_identity", Value: yaml.MapSlice{
 						{Key: "client_id", Value: spec.AzureAD.ManagedIdentity.ClientID},
-					},
-				},
+					}},
+				)
+			}
+
+			if spec.AzureAD.OAuth != nil {
+				// Azure OAuth is only supported in Prometheus 2.48.0 and later.
+				// The passed config might be invalid if the Prometheus version is older
+				// where neither the managed identity nor the oauth config get set.
+				if !cg.WithMinimumVersion("2.48.0").IsCompatible() {
+					return yaml.MapItem{}, fmt.Errorf("Azure OAuth is only supported in Prometheus 2.48.0 and later")
+				}
+				if spec.AzureAD.OAuth.ClientID == "" {
+					return yaml.MapItem{}, fmt.Errorf("clientId must be provided in the Azure AD OAuth config")
+				}
+				if spec.AzureAD.OAuth.ClientSecret == nil {
+					return yaml.MapItem{}, fmt.Errorf("clientSecret must be provided in the Azure AD OAuth config")
+				}
+				if spec.AzureAD.OAuth.TenantID == "" {
+					return yaml.MapItem{}, fmt.Errorf("tenantId must be provided in the Azure AD OAuth config")
+				}
+				_, err := uuid.Parse(spec.AzureAD.OAuth.ClientID)
+				if err != nil {
+					return yaml.MapItem{}, fmt.Errorf("the provided Azure OAuth clientId is invalid")
+				}
+				matched := tenantIDre.MatchString(spec.AzureAD.OAuth.TenantID)
```

Comment:
I'm a bit torn if we should validate this on the operator or not, because if MS change this we have to change the operator too. 

But!
If we think it's valuable, I'd prefer to add this annotation `// +kubebuilder:validation:Pattern` to the struct and let the validation on the CRD side.

WDYT?

Reply:
Moved validation to the CRD side. I'd keep it probably since prometheus itself does the same validation? Invalid id -> prometheus startup issues?

I'm fine with either though!